### PR TITLE
WIP: Display error widget in inspector page

### PR DESCRIPTION
**Pull request by albertusangga**
- Very early prototype to improve layout debugging process
- For this to work, this changes to the internal Flutter is also required to land:
https://github.com/albertusangga/flutter/tree/widget-inspector-to-support-layout-debugging

- Related to: #1044

### CODE REVIEW

File: packages/devtools_app/lib/src/inspector/inspector_tree.dart [hunk 1]
```diff
@@ -204,6 +204,13 @@ abstract class InspectorTreeNode {
     }
     final String name = diagnostic.name;
     TextStyle textStyle = textStyleForLevel(diagnostic.level);
+    if (diagnostic.isError) {
+      if (diagnostic.distanceFromError <= 0) {
+        textStyle = TextStyle(color: Colors.red, fontWeight: FontWeight.bold);
```
Comment (jacob314) [inline packages/devtools_app/lib/src/inspector/inspector_tree.dart]: lets move these TextStyle objects to the file that has the textStyleForLevel constants.

We'll also need to make sure the colors work ok in light and dark mode. I believe adding &theme=dark at the end of the URL will trigger dark mode.
Dark mode is the default when launching devtools from VSCode.

File: packages/devtools_app/lib/src/inspector/diagnostics_node.dart [hunk 2]
```diff
@@ -90,6 +90,12 @@ class RemoteDiagnosticsNode extends DiagnosticableTree {
   /// JSON describing the diagnostic node.
   final Map<String, Object> json;
 
+  // Describing whether the Node is an ErrorWidget
+  bool get isError => json['error'] == true;
+
+  // TODO(albertusangga): Comment
+  int get distanceFromError  => json['distanceFromError'];
```
Comment (jacob314) [inline packages/devtools_app/lib/src/inspector/diagnostics_node.dart]: Here is another idea to make the multiple errors case understandable to users:
We could make the InspectorService keep track of the concept of the "selected error" just like it tracks the concept of the selected widget.
Then we can highlight the selected error in red and other errors in orange.
The selected error would default to the first error returned but users could click around in the error logging view to select a different error or deselect all errors.

File: packages/devtools_app/web/index.html [hunk 3]
```diff
@@ -85,7 +85,6 @@
 </head>
 
 <body fullbleed layout vertical>
-    <div id="legacy-dart">
```
Comment (kenzieschmoll) [inline packages/devtools_app/web/index.html]: was removing this intentional? I thought we need this for a recent CL @DaveShuckerow landed?

Comment (jacob314) [reply packages/devtools_app/web/index.html]: Yes this is intentional. Dave's idea was good but unfortunately our CSS is a bit too fragile to support it. We need to verify that the flutter web version of the app works with this change.
If there is a problem, we will need to do a hybrid solution like tagging every top level div under body with class="legacy-dart" and then removing all those elements.

Comment (kenzieschmoll) [reply packages/devtools_app/web/index.html]: sg 👍

**Comment by albertusangga** [comment]
Closing this since we are writing Flutter code now, will open new PR soon! :)

**Pull request closed without merging**